src/Raven.Server/Documents/Handlers/Processors/Documents/AbstractDocumentHandlerProcessorForGet.cs

```diff
@@ -0,0 +1,451 @@
+            }
+
+            if (responseWriteStats != NoResults)
+            {
+                AddPagingPerformanceHint(PagingOperationType.Documents, actionName, HttpContext.Request.QueryString.Value, responseWriteStats.NumberOfResults,
+                    pageSize, sw.ElapsedMilliseconds, responseWriteStats.TotalDocumentsSizeInBytes);
+            }
+        }
+    }
+
+    protected async ValueTask<(long NumberOfResults, long TotalDocumentsSizeInBytes)> GetDocumentsByIdAsync(TOperationContext context, StringValues ids, StringValues includePaths, RevisionIncludeField revisions,
+        StringValues counters, HashSet<AbstractTimeSeriesRange> timeSeries, StringValues compareExchangeValues, bool metadataOnly, string etag)
+    {
+        var result = await GetDocumentsByIdImplAsync(context, ids, includePaths, revisions, counters, timeSeries, compareExchangeValues, metadataOnly, etag);
+
+        if (result.StatusCode == HttpStatusCode.NotFound)
+        {
+            if (etag == HttpCache.NotFoundResponse)
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+            else
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+
+            return NoResults;
+        }
+
+        if (etag == result.Etag)
+        {
+            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+
+            return NoResults;
+        }
+
+        HttpContext.Response.Headers[Constants.Headers.Etag] = "\"" + result.Etag + "\"";
+
+        if (SupportsHandlingOfMissingIncludes)
+            await HandleMissingIncludes(context, result, metadataOnly);
+
+        return await WriteDocumentsByIdResultAsync(context, metadataOnly, result);
+    }
+
+    private async ValueTask<(long NumberOfResults, long TotalDocumentsSizeInBytes)> WriteDocumentsByIdResultAsync(
+        TOperationContext context, bool metadataOnly, DocumentsByIdResult<TDocumentType> result)
+    {
+        long numberOfResults;
+        long totalDocumentsSizeInBytes;
+        await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+        {
+            writer.WriteStartObject();
+
+            (numberOfResults, totalDocumentsSizeInBytes) = await WriteDocumentsAsync(writer, context, nameof(GetDocumentsResult.Results), result.Documents, metadataOnly, CancellationToken);
+            
+            writer.WriteComma();
+
+            string includesPropertyName = nameof(GetDocumentsResult.Includes);
+
+            if (result.Includes.Count > 0)
+            {
+                await WriteIncludesAsync(writer, context, includesPropertyName, result.Includes, CancellationToken);
+            }
+            else
+            {
+                writer.WritePropertyName(includesPropertyName);
+                writer.WriteStartObject();
+                writer.WriteEndObject();
+            }
```

Comment:
this seems to be redundant and simply 
`await WriteIncludesAsync(writer, context, includesPropertyName, result.Includes, CancellationToken);` 
should be enough, no?

Reply:
Correct. Fixed.